base/dataset.go

```diff
@@ -228,9 +227,15 @@ func ListDatasets(ctx context.Context, r repo.Repo, term, profileID string, offs
 	return matches, nil
 }
 
+// Lister has the ability to list dsref.VersionInfos of a single user
+type Lister interface {
+	// List the dsref.VersionInfos of a single user
+	List(ctx context.Context, pid profile.ID, lp params.List) ([]dsref.VersionInfo, error)
+}
+
 // RawDatasetRefs converts the dataset refs to a string
-func RawDatasetRefs(ctx context.Context, pid profile.ID, s collection.Set) (string, error) {
-	res, err := s.List(ctx, pid, params.ListAll)
+func RawDatasetRefs(ctx context.Context, pid profile.ID, l Lister) (string, error) {
+	res, err := l.List(ctx, pid, params.ListAll)
```

Comment:
there was a import cycle when importing `transform` into `collection`. 

Narrowing down the interface and adding it to `base` seemed the best way to go.